packages/ui/react-ui-editor/src/extensions/comments.ts

```diff
@@ -545,17 +544,28 @@ export const scrollThreadIntoView = (view: EditorView, id: string, center = true
   if (!comment?.comment.cursor) {
     return;
   }
-
   const range = Cursor.getRangeFromCursor(view.state, comment.comment.cursor);
   if (range) {
-    view.dispatch({
-      selection: { anchor: range.from },
-      effects: [
-        //
-        EditorView.scrollIntoView(range.from, center ? { y: 'center' } : undefined),
-        setSelection.of({ current: id }),
-      ],
-    });
+    const currentSelection = view.state.selection.main;
+    const currentScrollPosition = view.scrollDOM.scrollTop;
+    const targetScrollPosition = view.coordsAtPos(range.from)?.top;
+
+    const needsScroll =
+      targetScrollPosition !== undefined &&
+      (targetScrollPosition < currentScrollPosition ||
+        targetScrollPosition > currentScrollPosition + view.scrollDOM.clientHeight);
```

Comment:
wondering if we should have something slightly smaller than the full client height in case the thing is right at the bottom then you might still want to scroll a bit? maybe just leave as a todo...

Reply:
I think it's fine, it leaves the comments in the middle of the view when scrolling.